receiver/splunkhecrawreceiver/receiver.go

```diff
@@ -0,0 +1,263 @@
+	ln, err := r.config.HTTPServerSettings.ToListener()
+	if err != nil {
+		return fmt.Errorf("failed to bind to address %s: %w", r.config.Endpoint, err)
+	}
+
+	mx := mux.NewRouter()
+	mx.NewRoute().HandlerFunc(r.handleReq)
+
+	r.server = r.config.HTTPServerSettings.ToServer(mx)
+
+	// TODO: Evaluate what properties should be configurable, for now
+	//		set some hard-coded values.
+	r.server.ReadHeaderTimeout = defaultServerTimeout
+	r.server.WriteTimeout = defaultServerTimeout
+	r.resourceCustomizer = r.createResourceCustomizer()
+
+	go func() {
+		if errHTTP := r.server.Serve(ln); errHTTP != http.ErrServerClosed {
+			host.ReportFatalError(errHTTP)
+		}
+	}()
+
+	return err
+}
+
+// Shutdown tells the receiver that should stop reception,
+// giving it a chance to perform any necessary clean-up.
+func (r *splunkReceiver) Shutdown(context.Context) error {
+	return r.server.Close()
+}
+
+func (r *splunkReceiver) handleReq(resp http.ResponseWriter, req *http.Request) {
+	ctx := req.Context()
+	if r.logsConsumer == nil {
+		ctx = r.obsrecv.StartMetricsOp(ctx)
+	}
+	reqPath := req.URL.Path
+	if !r.config.pathGlob.Match(reqPath) {
+		r.failRequest(ctx, resp, http.StatusNotFound, notFoundRespBody, nil)
+		return
+	}
+
+	if req.Method != http.MethodPost {
+		r.failRequest(ctx, resp, http.StatusBadRequest, invalidMethodRespBody, nil)
+		return
+	}
+
+	encoding := req.Header.Get(httpContentEncodingHeader)
+	if encoding != "" && encoding != gzipEncoding {
+		r.failRequest(ctx, resp, http.StatusUnsupportedMediaType, invalidEncodingRespBody, nil)
+		return
+	}
+
+	bodyReader := req.Body
+	if encoding == gzipEncoding {
+		var err error
+		bodyReader, err = gzip.NewReader(bodyReader)
+		if err != nil {
+			r.failRequest(ctx, resp, http.StatusBadRequest, errGzipReaderRespBody, err)
+			return
+		}
+	}
+
+	if req.ContentLength == 0 {
+		resp.Write(okRespBody)
+		return
+	}
+
+	sc := bufio.NewScanner(bodyReader)
+
+	ld := pdata.NewLogs()
+	rl := ld.ResourceLogs().AppendEmpty()
+	r.resourceCustomizer(req, rl.Resource())
+	ill := rl.InstrumentationLibraryLogs().AppendEmpty()
+
+	for sc.Scan() {
+		logRecord := ill.Logs().AppendEmpty()
+		logLine := sc.Text()
+		logRecord.Body().SetStringVal(logLine)
+	}
+	decodeErr := r.logsConsumer.ConsumeLogs(ctx, ld)
+
+	if decodeErr != nil {
+		r.failRequest(ctx, resp, http.StatusInternalServerError, errInternalServerError, decodeErr)
+	} else {
+		resp.WriteHeader(http.StatusAccepted)
+		resp.Write(okRespBody)
+	}
+}
+
+func (r *splunkReceiver) failRequest(
+	ctx context.Context,
+	resp http.ResponseWriter,
+	httpStatusCode int,
+	jsonResponse []byte,
+	err error,
+) {
+	resp.WriteHeader(httpStatusCode)
+	if len(jsonResponse) > 0 {
+		// The response needs to be written as a JSON string.
+		_, writeErr := resp.Write(jsonResponse)
+		if writeErr != nil {
+			r.logger.Warn("Error writing HTTP response message", zap.Error(writeErr))
+		}
+	}
+
+	msg := string(jsonResponse)
+
+	reqSpan := trace.FromContext(ctx)
```

Comment:
Where'd this span come from? I might not know about it, does the collector automatically create spans via opencensus?

Reply:
This was lifted from signalfxreceiver to do the same thing. Both signalfxreceiver and splunkhecreceiver handle this. The span is created by `ctx = r.obsrecv.StartLogsOp(ctx)` line 146.

Reply:
Thanks - if it works it's ok. I'm not sure if these spans actually have a chance of going anywhere though in the otel collector. If they can't actually be exported, I'd remove the trace related code here completely.

Reply:
I think this has to do with: https://github.com/open-telemetry/opentelemetry-collector/blob/main/obsreport/doc.go but we're doing something custom here instead of calling the `endSpan` method in their package. We could remove the extra attributes and that would allow us to use that DSL cleanly. I'll try that.

Reply:
Should be good there now.